### PR TITLE
Transfer liquidity check

### PR DESCRIPTION
**Pull request by DmitriyMolch**
https://cerenetwork.atlassian.net/browse/CBI-2020

### CODE REVIEW

File: src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx [hunk 1]
```diff
@@ -64,12 +62,14 @@ export const getBridgeProposalVotes = async (
   decimalAmount: number
 ): Promise<GetBridgeProsalVotesRes | undefined> => {
   const destinationChainConfig = getСhainConfig(destinationChainId);
```
Comment (DmitriyMolch) [inline src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx:64]: @AndreiNavoichyk , can we remove the config from this lib `ChainBridgeAPI`? It shouldn't know anything about app config.

File: config/chainbridge-runtime-config.cere.dev.js [hunk 2]
```diff
@@ -61,13 +61,15 @@ window.__RUNTIME_CONFIG__ = {
...
         existentialDepositPlusNetworkFee: "1.03",
+        bridgeAccountId: "5EYCAe5g7bGpFHagwe26HiRHdHdE3hobrwV6hq1UD2BPAiZb",
         tokens: [
           {
             address: "cere-native",
             name: cereTokenName,
             symbol: cereTokenSymbol,
             imageUri: "CEREIcon",
-            resourceId: "cere-native",
+            resourceId:
```
Comment (AndreiNavoichyk) [inline config/chainbridge-runtime-config.cere.dev.js:71]: Do you use it somewhere?

File: src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx (hunk 1 above)
Comment (AndreiNavoichyk) [reply src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx:64]: @Dmitriy1984Molchanov 
Sure. Please, do it in a separate PR.

File: src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx [hunk 3]
```diff
@@ -83,10 +83,36 @@ export const getBridgeProposalVotes = async (
...
+  );
+};
+
+export const hasTokenSupplies = async (
+  api: ApiPromise,
+  address: string,
+  amount: number,
+  decimals: number
+): Promise<boolean> => {
+  const balance = await getBalance(api, address);
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx:108]: What will be if the request fails?

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx [hunk 4]
```diff
@@ -262,35 +282,19 @@ export const EVMHomeAdaptorProvider = ({
...
-            token,
-            amount,
-            tokenAddress
-          );
-          if (!hasSupplies) {
-            return false;
-          }
-        }
-        return true;
+        if (!hasSupplies) return false;
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx]: Just return `hasSupplies`?

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx [hunk 5]
```diff
@@ -262,35 +282,19 @@ export const EVMHomeAdaptorProvider = ({
...
       }
+      return true;
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx]: if `destinationChainConfig?.type` is not `Substrate` we should log error and return `false`.

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx [hunk 6]
```diff
@@ -297,6 +298,36 @@ export const SubstrateHomeAdaptorProvider = ({
...
+            tokens,
+            token,
+            amount,
+            tokenAddress
+          );
+          if (!hasSupplies) {
+            return false;
+          }
+        }
+        return true;
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: if `destinationChain?.type` is not `Ethereum` we should log error and return `false`.

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx [hunk 7]
```diff
@@ -297,6 +298,36 @@ export const SubstrateHomeAdaptorProvider = ({
...
+
+        if (destinationChain?.type === "Ethereum" && token) {
+          const hasSupplies = await hasTokenSupplies(
+            destinationChain,
+            tokens,
+            token,
+            amount,
+            tokenAddress
+          );
+          if (!hasSupplies) {
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: Just return `hasSupplies`

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx [hunk 8]
```diff
@@ -262,35 +282,19 @@ export const EVMHomeAdaptorProvider = ({
...
   const handleCheckSupplies = useCallback(
-    async (
-      amount: number,
-      tokenAddress: string,
-      destinationChainId: number
-    ) => {
-      if (homeChainConfig) {
-        const destinationChain = chainbridgeConfig.chains.find(
-          (c) => c.chainId === destinationChainId
+    async (amount: number) => {
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx:285]: How often will the check execute?

File: config/chainbridge-runtime-config.cere.dev.js (hunk 2 above)
Comment (DmitriyMolch) [reply config/chainbridge-runtime-config.cere.dev.js:71]: Yes, we use it here https://github.com/Cerebellum-Network/chainbridge-ui/blob/c167d29a7f14ba3a26970833397885c63b8791cb/src/Contexts/Adaptors/EVMAdaptors/helpers.ts#L61

File: src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx (hunk 3 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/SubstrateApis/ChainBridgeAPI.tsx:108]: The same thing as in the case  `deposit` fails. It should take some common error handler.

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx (hunk 8 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx:285]: Once we clicked the "Start Transfer" button

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx (hunk 4 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx]: fixed

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx (hunk 5 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx]: fixed

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx (hunk 6 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: fixed

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx (hunk 7 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: fixed

File: src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx [hunk 9]
```diff
@@ -262,35 +282,22 @@ export const EVMHomeAdaptorProvider = ({
...
+          api as ApiPromise,
+          (destinationChainConfig as SubstrateBridgeConfig).bridgeAccountId,
+          amount,
+          destinationChainConfig.decimals
         );
-        const token = homeChainConfig.tokens.find(
-          (token) => token.address === tokenAddress
+      } else {
+        console.warn(
+          `The destination chain type ${destinationChainConfig?.type} is unknown. Please check it.`
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx]: Add in the beginning `Liquidity check is skipping.`

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx [hunk 10]
```diff
@@ -297,6 +298,37 @@ export const SubstrateHomeAdaptorProvider = ({
...
+          return await hasTokenSupplies(
+            destinationChainConfig,
+            tokens,
+            token,
+            amount,
+            tokenAddress
+          );
+        } else {
+          console.warn(
+            `The destination chain type ${destinationChainConfig?.type} is unknown. Please check it.`
```
Comment (AndreiNavoichyk) [inline src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: Add in the beginning `Liquidity check is skipping.`

**Review by AndreiNavoichyk — APPROVED** [review]

File: src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx (hunk 10 above)
Comment (DmitriyMolch) [reply src/Contexts/Adaptors/SubstrateHomeAdaptor.tsx]: fixed

**Pull request merged**